spatialmath/box.go

```diff
@@ -14,26 +14,28 @@ import (
 type boxCreator struct {
 	halfSize r3.Vector
 	pointCreator
+	label string
 }
 
 // box is a collision geometry that represents a 3D rectangular prism, it has a pose and half size that fully define it.
 type box struct {
 	pose     Pose
 	halfSize [3]float64
+	label    string
 }
 
 // NewBoxCreator instantiates a BoxCreator class, which allows instantiating boxes given only a pose which is applied
 // at the specified offset from the pose. These boxes have dimensions given by the provided halfSize vector.
-func NewBoxCreator(dims r3.Vector, offset Pose) (GeometryCreator, error) {
+func NewBoxCreator(dims r3.Vector, offset Pose, label string) (GeometryCreator, error) {
 	if dims.X <= 0 || dims.Y <= 0 || dims.Z <= 0 {
 		return nil, newBadGeometryDimensionsError(&box{})
 	}
-	return &boxCreator{dims.Mul(0.5), pointCreator{offset}}, nil
+	return &boxCreator{dims.Mul(0.5), pointCreator{offset, label}, label}, nil
 }
 
 // NewGeometry instantiates a new box from a BoxCreator class.
 func (bc *boxCreator) NewGeometry(pose Pose) Geometry {
```

Comment:
@raybjork  Do you think the new label argument go in the signature of BoxCreator, or in the signature of NewGeometry?

Reply:
Yea, I wondering if it's *only* NewGeometry() that should pass the label to the box{}/sphere{}/etc struct, or if it should be xCreator as you currently have it. I'll want to wait to hear from @raybjork to see how they use these creator, and if it makes sense to set the label at the creator level.

Reply:
The GeometryCreator is the one that should have the label and provide it.

The way this information is used, is that any individual geometry is intended to be ephemeral and only valid at that location. The GeometryCreator persists the information about things like the dimensions of the geometry, and instantiates new versions of that geometry at various poses as requested.

There shouldn't be a situation where people are calling NewGeometry on a GeometryCreator and wanting it to have a different label, that should be an entirely new GeometryCreator.

Reply:
Excellent, then all the open questions are resolved on my end- LGTM @bazile-clyde